commands/show_test.go

```diff
@@ -59,48 +71,69 @@ func TestBlockDaemon(t *testing.T) {
 func TestShowDeal(t *testing.T) {
 	tf.IntegrationTest(t)
 
-	minerDaemon := th.NewDaemon(t,
-		th.WithMiner(fixtures.TestMiners[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[0]),
-		th.DefaultAddress(fixtures.TestAddresses[0]),
-		th.AutoSealInterval("1"),
-	).Start()
-	defer minerDaemon.ShutdownSuccess()
+	fastenvOpts := fast.EnvironmentOpts{
+		//InitOpts: []fast.ProcessInitOption{fast.POAutoSealIntervalSeconds(1)},
+		//DaemonOpts: []fast.ProcessDaemonOption{fast.POBlockTime(time.Second)},
+	}
 
-	clientDaemon := th.NewDaemon(t,
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-		th.DefaultAddress(fixtures.TestAddresses[1]),
-	).Start()
-	defer clientDaemon.ShutdownSuccess()
+	ctx, env := fastesting.NewTestEnvironment(context.Background(), t, fastenvOpts)
 
-	minerDaemon.RunSuccess("mining", "start")
-	minerDaemon.UpdatePeerID()
+	clientNode := env.GenesisMiner
+	require.NoError(t, clientNode.MiningStart(ctx))
 
-	minerDaemon.ConnectSuccess(clientDaemon)
+	minerNode := env.RequireNewNodeWithFunds(1000)
+	defer func() {
+		require.NoError(t, env.Teardown(ctx))
+	}()
 
-	addAskCid := minerDaemon.MinerSetPrice(fixtures.TestMiners[0], fixtures.TestAddresses[0], "20", "10")
+	// Connect the clientNode and the minerNode
+	err := series.Connect(ctx, clientNode, minerNode)
+	require.NoError(t, err)
 
-	t.Run("when deal exists shows deal info", func(t *testing.T) {
-		clientDaemon.WaitForMessageRequireSuccess(addAskCid)
-		dataCid := clientDaemon.RunWithStdin(strings.NewReader("HODLHODLHODL"), "client", "import").ReadStdoutTrimNewlines()
+	// Create a minerNode
+	collateral := big.NewInt(500)           // FIL
+	price := big.NewFloat(0.000000001)      // price per byte/block
+	expiry := big.NewInt(24 * 60 * 60 / 30) // ~24 hours
 
-		proposeDealOutput := clientDaemon.RunSuccess("client", "propose-storage-deal", fixtures.TestMiners[0], dataCid, "0", "5").ReadStdoutTrimNewlines()
+	ask, err := series.CreateStorageMinerWithAsk(ctx, minerNode, collateral, price, expiry)
+	require.NoError(t, err)
 
-		splitOnSpace := strings.Split(proposeDealOutput, " ")
+	// Create some data that is the full sector size and make it autoseal asap
+	var data bytes.Buffer
```

Comment:
do we end up using this for anything, at first glance this looks like its not required.

Reply:
I got it from another test -- I think it was to cause autoseal to happen.

Reply:
I believe it was there just to tee the data off into so we could keep a copy of the random data around to assert on later. I don't believe you need to to call `io.TeeReader` here as I don't believe the test actually cares about the contents of the piece being stored.